F/690-improve-equalizer

kathwill opened the pull request:
# Resolves #690

**Does your pull request solve a problem? Please describe:**  
The tab for the equalizer is not self descriptive. The pull request changes the name of the tab and adds explanations for the shown numbers.

**Affected Component(s):**  
`Frontend`

**Additional context:**  
<img width="902" alt="QueryWeights" src="https://user-images.githubusercontent.com/49514526/87544844-62f0ae80-c6a7-11ea-8151-cf07b08c2307.png">

<img width="902" alt="QueryWeights3" src="https://user-images.githubusercontent.com/49514526/88376048-fc514c00-cd9c-11ea-9f85-e6ba41e2a03d.png">

Commit:
change equalizer to query weights

kathwill committed:
Merge branch 'dev' into F/690-improve-equalizer

Commit:
add text for query numbers and weights

kathwill committed:
Merge branch 'dev' into F/690-improve-equalizer

kathwill committed:
Merge branch 'dev' into F/690-improve-equalizer

kathwill commented:
What do you think of this option? @Bensk1 @Bouncner

Bensk1 commented:
We are getting there!

- I am not a UI expert but what about putting the descriptions _query numbers_ and _query weights_ on the left side?
- Could we also rename _Query numbers_ to _Query ID_ and _Query Weights_ to _Query Weight_?
- Also, while we are on it. Do you think that a reset button (including functionality) is complicated?
- Does anyone have other UI/UX ideas? A different font/color to better differentiate between text labels/control elements?

Bouncner commented:
Is it possible to change the input fields style to make it more obvious that this is an editable field and not just text?

Commit:
rename labels

Commit:
Merge branch 'F/690-improve-equalizer' of github.com:hyrise/Cockpit into F/690-improve-equalizer

kathwill committed:
Merge branch 'dev' into F/690-improve-equalizer

Commit:
add reset button for weights

kathwill commented:
I renamed _Query numbers_ to _Query ID_ and _Query Weights_ to _Query Weight_ and added a reset button (as you can see in the pictures above).

> I am not a UI expert but what about putting the descriptions _query numbers_ and _query weights_ on the left side?

I decided to put it on the right side because I think if we put it on the left side it is too much information on that side because of the scale. What do you think of this @Bensk1?

Bensk1 commented:
> I decided to put it on the right side because I think if we put it on the left side it is too much information on that side because of the scale. What do you think of this @Bensk1?

Perfect, I like that better.

schTi commented on hyrisecockpit/frontend/src/components/workload/QueryWeights.vue:
```diff
@@ -52,8 +52,41 @@
               />
             </div>
+            <div
+              v-if="
+                idx % numberOfQueriesPerRow === numberOfQueriesPerRow - 1 ||
+                idx ===
+                  numberOfQueriesPerWorkload[
+                    selectedWorkloads.indexOf(workload)
+                  ] -
+                    1
+              "
```
looks like this logic can be outsourced in an own function

schTi commented on hyrisecockpit/frontend/src/components/workload/QueryWeights.vue:
```diff
@@ -129,6 +167,15 @@ export default defineComponent({
+      Object.values(weights.value[workload]).forEach((weight) => {
+        if (weight.value !== 1) {
+          weight.value = 1;
+          weight.sliderValue = 50;
```
can you add a comment, why 50 is the base value?

Commit:
add comments

Commit:
rename constants

Commit:
add function for logic

kathwill committed:
Merge branch 'dev' into F/690-improve-equalizer

schTi committed:
Merge branch 'dev' into F/690-improve-equalizer
